ui/DesignSystem/Component/SourceBrowser/RevisionSelector.svelte

```diff
@@ -1,19 +1,132 @@
 <script>
   import { createEventDispatcher } from "svelte";
 
-  import { Input } from "../../Primitive";
+  import { Avatar, Icon } from "../../Primitive";
 
+  export let currentRevision = null;
+  export let expanded = false;
   export let revisions = null;
+  export let style = "";
 
-  let revision;
+  // Dropdown element. Set by the view.
+  let dropdown = null;
+
+  const showDropdown = () => {
+    expanded = true;
+  };
+
+  const hideDropdown = () => {
+    expanded = false;
+  };
+
+  const handleClick = ev => {
+    if (dropdown !== ev.target && !dropdown.contains(ev.target)) {
```

Comment:
Why do we need these checks?
It seems to work correctly even if I remove them.

Reply:
If you click inside the dropdown when expanded, eg. on the user handle, it will hide the dropdown.

Reply:
I see, I was just clicking on the branch and `on:click|stopPropagation` was making sure that worked. Didn't think about clicking around elsewhere in the menu :)